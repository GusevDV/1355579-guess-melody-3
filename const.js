export const QuestionTypes = {
  ARTIST: `artist`,
  GENRE: `genre`,
};
