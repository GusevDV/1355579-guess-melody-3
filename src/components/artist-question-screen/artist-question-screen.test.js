import React from 'react';
import renderer from 'react-test-renderer';
import ArtistQuestionScreen from './artist-question-screen.jsx';
import {QuestionTypes} from '../../../const.js';

const question = {
  type: QuestionTypes.ARTIST,
  song: {
    artist: `Loboda`,
    src: `https://store.vas-stream.ru/uploads/mts-rbt/cms/preview/file/15336/319fce3a4c1668930a0b24a2ceac7fa2.optimized.mp3`,
  },
  answers: [{
    picture: `https://igapi.megafon.ru/static/thumbs/singers/Loboda/244x244`,
    artist: `Loboda`,
  }, {
    picture: `https://igapi.megafon.ru/static/thumbs/singers/Zivert/244x244`,
    artist: `Zivert`,
  }, {
    picture: `https://store.vas-stream.ru/uploads/mts-rbt/cms/image/file/7906/optimized_default.jpg`,
    artist: `KAZKA`,
  }],
};

it(`ArtistQuestionScreen is rendered correctly`, () => {
  const tree = renderer.create(
      <ArtistQuestionScreen
        question={question}
        onAnswer={() => {}}
        renderPlayer={() => {}}
      />,
      {
        createNodeMock: () => {
          return {};
        }
      }).toJSON();

  expect(tree).toMatchSnapshot();
});
