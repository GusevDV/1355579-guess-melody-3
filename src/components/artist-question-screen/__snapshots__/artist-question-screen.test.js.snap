// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionScreen is rendered correctly 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
        style={
          Object {
            "filter": "url(#blur)",
            "transform": "rotate(-90deg) scaleY(-1)",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio
            src="https://store.vas-stream.ru/uploads/mts-rbt/cms/preview/file/15336/319fce3a4c1668930a0b24a2ceac7fa2.optimized.mp3"
          />
        </div>
      </div>
    </div>
    <form
      className="game__artist"
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-0"
          name="answer"
          onChange={[Function]}
          type="radio"
          value="artist-0"
        />
        <label
          className="artist__name"
          htmlFor="answer-0"
        >
          <img
            alt="Loboda"
            className="artist__picture"
            src="https://igapi.megafon.ru/static/thumbs/singers/Loboda/244x244"
          />
          Loboda
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-1"
          name="answer"
          onChange={[Function]}
          type="radio"
          value="artist-1"
        />
        <label
          className="artist__name"
          htmlFor="answer-1"
        >
          <img
            alt="Zivert"
            className="artist__picture"
            src="https://igapi.megafon.ru/static/thumbs/singers/Zivert/244x244"
          />
          Zivert
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-2"
          name="answer"
          onChange={[Function]}
          type="radio"
          value="artist-2"
        />
        <label
          className="artist__name"
          htmlFor="answer-2"
        >
          <img
            alt="KAZKA"
            className="artist__picture"
            src="https://store.vas-stream.ru/uploads/mts-rbt/cms/image/file/7906/optimized_default.jpg"
          />
          KAZKA
        </label>
      </div>
    </form>
  </section>
</section>
`;
